cstar/base/input_dataset.py

```diff
@@ -88,6 +75,14 @@ def __init__(
         assert self.start_date is None or isinstance(self.start_date, dt.datetime)
         assert self.end_date is None or isinstance(self.end_date, dt.datetime)
 
+    @property
+    def exists_locally(self) -> bool:
+        return (
+            True
+            if ((self.working_path is not None) and (self.working_path.exists()))
```

Comment:
Pretty sure most, if not all of these brackets are redundant
```suggestion
            if (self.working_path is not None) and self.working_path.exists()
```

Reply:
They are, it's personal preference as most editors will highlight the matching bracket when you hover on one - makes it easier to see the nested structure / exactly what each condition is in chains of `and`s and `or`s